.github/workflows/main.yml

```diff
@@ -87,9 +87,52 @@ jobs:
           name: coverage-${{ matrix.python-version }}
           path: coverage
 
+  test-integration:
```

Comment:
I think having a different job is a way to "motivate" us to write more tests like those, since it's going to eventually slow down us.

Reply:
Your complaint seems more about the amount of duplicated code, rather than the idea of having a separate flow. Which can be solved by creating a common GH action.